app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -30,21 +29,17 @@ public int layout() {
 
     @Override
     public void init() {
-
         ButterKnife.bind(this, getActivity());
     }
 
-
     @Override
     public void setListeners() {
-
         vLogInButton.setOnClickListener(new View.OnClickListener() {
             @OnClick
             public void onClick(View v) {
-                if (areValidFields())
+                if (getPresenter().ValidationFields(vLogInEmail.getText().toString(), vLogInPassword.getText().toString()))
```

Comment:
It is bad, but not that bad xD I would do something like: getPresenter.ValidationFields(fafafa). Then in the presenter: validationFields(){ if ( itIsEverythingOK) doLogin(), else getView().showError.
Also remember that functions names are in camelCase and not in PascalCase. Always begins in small letter.